processor/resourcedetectionprocessor/internal/azure/aks/aks.go

```diff
@@ -62,7 +67,38 @@ func onK8s() bool {
 	return os.Getenv(kubernetesServiceHostEnvVar) != ""
 }
 
-func azureMetadataAvailable(ctx context.Context, p azure.Provider) bool {
-	_, err := p.Metadata(ctx)
-	return err == nil
+// parseClusterName parses the cluster name from the infrastructure
+// resource group name. AKS IMDS returns the resource group name in
+// the following formats:
+//
+// 1. Generated group: MC_<resource group>_<cluster name>_<location>
+//   - Example:
+//   - Resource group: my-resource-group
+//   - Cluster name:   my-cluster
+//   - Location:       eastus
+//   - Generated name: MC_my-resource-group_my-cluster_eastus
+//
+// 2. Custom group: custom-infra-resource-group-name
+//
+// When using the generated infrastructure resource group, the resource
+// group will include the cluster name. If the cluster's resource group
+// or cluster name contains underscores, parsing will fall back on the
+// unparsed infrastructure resource group name.
+//
+// When using a custom infrastructure resource group, the resource group name
+// does not contain the cluster name. The custom infrastructure resource group
+// name is returned instead.
+//
+// It is safe to use the infrastructure resource group name as a unique identifier
+// because Azure will not allow the user to create multiple AKS clusters with the same
+// infrastructure resource group name.
+func parseClusterName(resourceGroup string) string {
+	// Code inspired by https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/datadogexporter/internal/hostmetadata/internal/azure/provider.go#L36
```

Comment:
Nice description for `parseClusterName` 👍 

Assuming that this code won't be removed from `exporter/datadogexporter` going forward (cc @mx-psi to confirm), we should consolidate it in `internal/azure` and use it both here and in `exporter/datadogexporter`. Moving it from `exporter/datadogexporter` to `internal/azure` can be done in a separate PR.

Reply:
We are not going to remove this from the Datadog exporter in the short term,so I agree we should consolidate this :+1:

Reply:
Cool, no action needed from me?

Reply:
So we'd like to make it a common utility in `internal/azure` and update the components, but can be done in a separate PR. @dmitryax if this is not a blocker, I think we are good to merge this.

Reply:
Sounds good to me as a follow-up PR. @jsirianni will you have a chance to submit that PR?

Reply:
@dmitryax I think I can handle that :+1:

Reply:
@jsirianni thanks. Submitted https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/30834